cmd/fpga_plugin/fpga_plugin.go

```diff
@@ -358,6 +380,38 @@ func getPluginParams(mode string) (getDevTreeFunc, string, error) {
 	return getDevTree, annotationValue, nil
 }
 
+func generateAndWriteCDISpec(deviceID, cdiPath string) error {
+	spec := cdispec.Spec{
+		Version: CDIVersion,
+		Kind:    CDIKind,
+		Devices: []cdispec.Device{
+			{
+				Name: deviceID,
+				ContainerEdits: cdispec.ContainerEdits{
+					Hooks: []*cdispec.Hook{
+						{
+							HookName: HookName,
+							Path:     HookPath,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	jsonSpec, err := json.Marshal(spec)
+	if err != nil {
+		return err
+	}
+
+	cdiFileName := path.Join(cdiPath, deviceID) + ".json"
+	if err = os.WriteFile(cdiFileName, jsonSpec, 0o644); err != nil { //nolint:gosec
+		return err
+	}
```

Comment:
Would this code benefit from usage of CDI APIs + workaround https://github.com/cncf-tags/container-device-interface/issues/224 (setting permissions).

Reply:
You tell me, the original code from you. :)

But in general, using the CDI APIs would remove the marshaling + manual writes. Maybe we could also exit-early if the spec is already available.

Reply:
I updated the code to use the cdi cache.